modules/dxp/apps/portal-security-audit/portal-security-audit-event-generators/src/main/java/com/liferay/portal/security/audit/event/generators/internal/security/auth/LoginDoesNotExist.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.portal.security.audit.event.generators.internal.security.auth;
+
+import com.liferay.portal.kernel.audit.AuditException;
+import com.liferay.portal.kernel.audit.AuditMessage;
+import com.liferay.portal.kernel.audit.AuditRouter;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.CompanyConstants;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.security.auth.AuthDoesNotExist;
+import com.liferay.portal.security.audit.event.generators.constants.EventTypes;
+
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Nikoletta Buza
+ */
+@Component(property = "key=auth.dne", service = AuthDoesNotExist.class)
+public class LoginDoesNotExist implements AuthDoesNotExist {
+
+	@Override
+	public void onUserDoesNotExist(
+		String authType, long companyId, String login,
+		Map<String, String[]> headerMap) {
```

Comment:
Since are now following the full pattern established in `LoginFailure` we should have separate methods for the 3 different auth types (emailAddress, screenName, userId).

Reply:
I need some clarification regarding the method separation so I can hopefully send a better commit!
If I understand correctly, you suggest we should change the method signature as well? Because otherwise it would be 3 identical methods, just with different names.

Some more things I noticed while I refactored the code and why I did not fully follow the LoginFailure pattern:
1. In case of LoginFailure, there are further operations after a failed login, unlike in case of a user does not exist outcome, that need the already authenticated, correct login, hence the separate methods. For example the user with the provided email/screen name/id is fetched from the DB and updated with the date of the failed login. Or in `LoginFailure`, the user is is fetched from the DB by emial/screen name/id to add user info to the audit message. In our case, there are no database operations because there is no user record with the incorrect login so I could not find a substantial reason to apply the same pattern here.
2. Furthermore, the logs would not be different either with this method signature.
3. Also, the LoginFailure has a method parameter that is never used (`String[]> parameterMap`), if you meant changing the method signature, would it be a good idea to follow that as well?

Can you please provide more details on this?

Reply:
Alright, thank you!

[3] I was not aware of the frontend using it, thanks!

However, `LoginFailure` methods have both a `headerMap` (used in the audit log itself) and a `parameterMap` (e.g. `public void onFailureByEmailAddress(
		long companyId, String emailAddress, Map<String, String[]> headerMap,
		Map<String, String[]> parameterMap)`)
		
From `UserLocalServiceImpl.authenticate`:
```
@param  headerMap the header map from the authentication request
@param  parameterMap the parameter map from the authentication request
```

I added only the `headerMap` because I did not use the latter explicitly in the method.


So considering all this, I should rather add the `parameterMap` as well, after the `headerMap`, if I understand correctly?

Reply:
You are right again. Sorry, am doing too much context switching today! :)
So I would add both and populate the `headerMap` using the `HttpServletRequest` as with the `LoginFailure` case.